From 4af0561e11ab24cdec6bba5fded79518a7a97358 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:23:40 -0400
Subject: [eisa] Call put_device() if device_register() fails

Message-id: <20140719042340.19438.42558.stgit@gir.stowe>
Patchwork-id: 85971
O-Subject: [RHEL7.1 PATCH 09/81] EISA: Call put_device() if device_register() fails
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 04df008afe0c569fd09400a35b7d3db29067114c

commit 04df008afe0c569fd09400a35b7d3db29067114c
Author: Levente Kurusa <levex@linux.com>
Date:   Fri Dec 13 19:39:54 2013 +0100

    EISA: Call put_device() if device_register() fails

    We need to give up the last reference to edev->dev, so we need to call
    put_device().

    Signed-off-by: Levente Kurusa <levex@linux.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/eisa/eisa-bus.c b/drivers/eisa/eisa-bus.c
index 272a3ec..8842cde 100644
--- a/drivers/eisa/eisa-bus.c
+++ b/drivers/eisa/eisa-bus.c
@@ -232,8 +232,10 @@ static int __init eisa_init_device(struct eisa_root_device *root,
 static int __init eisa_register_device(struct eisa_device *edev)
 {
  int rc = device_register(&edev->dev);
- if (rc)
+ if (rc) {
+  put_device(&edev->dev);
   return rc;
+ }
 
  rc = device_create_file(&edev->dev, &dev_attr_signature);
  if (rc)
-- 
1.7.1